packages/transaction-controller/src/utils/batch.ts

```diff
@@ -512,3 +540,35 @@ async function processTransactionWithHook(
     params: newParams,
   };
 }
+
+/**
+ * Requests approval for a transaction batch by interacting with the ApprovalController.
+ *
+ * @param txBatchMeta - Metadata for the transaction batch, including its ID and origin.
+ * @param messenger - The messenger instance used to communicate with the ApprovalController.
+ * @param options - Options for the approval request.
+ * @param options.shouldShowRequest - A boolean indicating whether the approval request should be displayed to the user.
+ * @returns A promise that resolves to the result of adding the approval request.
+ */
+async function requestApproval(
+  txBatchMeta: TransactionBatchMeta,
+  messenger: TransactionControllerMessenger,
+  { shouldShowRequest }: { shouldShowRequest: boolean },
+): Promise<AddResult> {
+  const id = String(txBatchMeta.id);
+  const { origin } = txBatchMeta;
+  const type = 'transaction_batch';
+  const requestData = { txBatchId: id };
+
+  return (await messenger.call(
+    'ApprovalController:addRequest',
+    {
+      id,
+      origin: origin || ORIGIN_METAMASK,
+      requestData,
+      expectsResult: true,
```

Comment:
If we're setting `expectsResult` to `true`, it means the client will wait until we call one of the result callbacks.

Meaning we need to invoke `success` or `error` returned from this function once the batch is complete or failed.

Reply:
So, should I save the `resultCallbacks` and call success when tx is confirmed or error when tx fails, similar to the pattern of `collectHook`?
Applied fixes.